src/Functions/FunctionsHashing.h

```diff
@@ -1369,6 +1370,35 @@ class FunctionURLHash : public IFunction
     }
 };
 
+struct ImplMeowHash128
+{
+    static constexpr auto name = "meowHash128";
+    using ReturnType = UInt128;
+
+    static UInt128 apply(const char *s, const size_t len)
+    {
+        union {
+            __m128i m128;
+            UInt128 u128;
+        };
+        m128 = MeowHash(MeowDefaultSeed, len, const_cast<char *>(s));
```

Comment:
I don't think const_cast is a good approach - we shouldn't rely on current implementation of the library. Your previous variant was better, but you can't use strdup here - 's' can contain zeroes ( `SELECT meowHash128('aaa\0bbb')` ). Consider something like:
```
std::string str(s, len);
m128 = MeowHash(MeowDefaultSeed, str.size(), str.data());
```

Reply:
It's not worth extra copying and (possibly) allocations.
Let's patch the library or use const_cast (which is ok - many C libraries are written without good constness).

Reply:
I believe GCC and clang both support __attribute__((__may_alias__)) - I already used it in another hash